src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -140,6 +145,183 @@ class_implements($class_name)
   /**
    * {@inheritdoc}
    */
+  public function value(DataSourceInterface $source) {
+    // Return if this field is a callback.
+    $value = $this->decorated->value($source);
+    if (isset($value)) {
+      return $value;
+    }
+
+    if ($resource = $this->getResource()) {
+      // TODO: The resource input data in the field definition has changed.
+      // Now it does not need to be keyed by bundle since you don't even need
+      // an entity to use the resource based field.
+
+      $request = Request::create('', array(), Request::METHOD_GET);
+      $resource_data_provider = DataProviderResource::init($request, $resource['name'], array(
+        $resource['majorVersion'],
+        $resource['minorVersion'],
+      ));
+      // FIXME: $embedded_identifier needs to be fetched first!!!
+      return $resource_data_provider->view($embedded_identifier);
+    }
+
+    // Check user has access to the property.
+    if (!$this->access('view', $source)) {
+      return NULL;
+    }
+
+    $property_wrapper = $this->propertyWrapper($source);
+    if ($this->getFormatter()) {
+      // Get value from field formatter.
+      $value = $this->formatterValue($source);
+    }
+    elseif ($property_wrapper instanceof \EntityListWrapper) {
+      // Multiple values.
+      foreach ($property_wrapper->getIterator() as $item_wrapper) {
+        $value[] = $this->fieldValue($item_wrapper);
+      }
+    }
+    else {
+      // Single value.
+      $value = $this->fieldValue($property_wrapper);
+    }
+
+    return $value;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \EntityMetadataWrapperException
+   */
+  public function access($op, DataSourceInterface $source) {
+    // Perform basic access checks.
+    if (!$this->decorated->access($op, $source)) {
+      return FALSE;
+    }
+
+    if (!$this->getProperty()) {
+      // If there is no property we cannot check for property access.
```

Comment:
Meaning that we can only check for field access or property access for things like `$file->filemime` or `$node->field_tags`. We denote those by using the `'property'` property in the plugin definition.